### PR TITLE
Refactored

### PR DESCRIPTION
**Pull request by supersecretliam**
* Added MarketName enum
* Changed profitAndLoss to funds
* Markets are dynamically generated by the list of enums

### CODE REVIEW

File: src/main/java/com/iggroup/universityworkshopmw/domain/model/Client.java [hunk 1]
```diff
@@ -15,6 +15,6 @@
    private String clientId;
    private String userName;
    @Setter
-   private double profitAndLoss;
+   private Double funds;
```
Comment (supersecretcatia) [inline src/main/java/com/iggroup/universityworkshopmw/domain/model/Client.java]: I made funds here double and not Double, as Double takes up more space (java is efficient etc I know). I thought Double, Integer etc are used primarily in DTOs, and use int, double elsewhere, unless there is some reason for it

File: src/main/java/com/iggroup/universityworkshopmw/domain/services/MarketDataService.java [hunk 2]
```diff
@@ -46,78 +47,17 @@ void updateMarket(Market market) {
...
-            .currentPrice(125.0)
-            .build());
-      //10
-      marketId = ID_PREFIX + ++marketIdNumber;
-      marketIdToMarketModelMap.put(marketId, Market.builder()
-            .marketId(marketId)
-            .marketName("Sugar")
-            .currentPrice(148.0)
-            .build());
+      IntStream.range(0, MarketName.values().length)
```
Comment (supersecretcatia) [inline src/main/java/com/iggroup/universityworkshopmw/domain/services/MarketDataService.java:50]: so much nicer looking than previously!

File: src/main/java/com/iggroup/universityworkshopmw/integration/transformers/MarketDataTransformer.java [hunk 3]
```diff
@@ -11,7 +11,7 @@
       return listOfMarkets.stream()
             .map(market -> MarketDto.builder()
                   .marketId(market.getMarketId())
-                  .marketName(market.getMarketName())
+                  .marketName(market.getMarketName().getMarketName())
```
Comment (supersecretcatia) [inline src/main/java/com/iggroup/universityworkshopmw/integration/transformers/MarketDataTransformer.java]: could this be changed to something like market.getMarketName.name()

File: src/main/java/com/iggroup/universityworkshopmw/domain/model/Client.java (hunk 1 above)
Comment (supersecretliam) [reply src/main/java/com/iggroup/universityworkshopmw/domain/model/Client.java]: Okay sure. I can change that

File: src/main/java/com/iggroup/universityworkshopmw/integration/transformers/MarketDataTransformer.java (hunk 3 above)
Comment (supersecretliam) [reply src/main/java/com/iggroup/universityworkshopmw/integration/transformers/MarketDataTransformer.java]: yeah, we could do that

File: src/main/java/com/iggroup/universityworkshopmw/domain/model/Client.java [hunk 4]
```diff
@@ -15,6 +15,6 @@
    private String clientId;
    private String userName;
    @Setter
-   private double profitAndLoss;
+   private funds;
```
Comment (supersecretcatia) [inline src/main/java/com/iggroup/universityworkshopmw/domain/model/Client.java]: private 'double' funds?

File: src/main/java/com/iggroup/universityworkshopmw/domain/model/Market.java [hunk 5]
```diff
@@ -12,7 +13,7 @@
 public class Market {
 
    private String marketId;
-   private String marketName;
+   private MarketName marketName;
    private Double currentPrice;
```
Comment (supersecretcatia) [inline src/main/java/com/iggroup/universityworkshopmw/domain/model/Market.java]: my bad - I had Double in here!! change to double

**Pull request merged**